src/meta/server_state.cpp

```diff
@@ -1438,14 +1439,17 @@ void server_state::recall_app(dsn::message_ex *msg)
 }
 
 void server_state::list_apps(const configuration_list_apps_request &request,
-                             configuration_list_apps_response &response)
+                             configuration_list_apps_response &response,
+                             dsn::message_ex *msg)
```

Comment:
how about mark this function as `const`? then use const in the loop in line 1447.